airflow/jobs/scheduler_job.py

```diff
@@ -1291,7 +1291,9 @@ def _schedule_dag_run(
             self.log.error("Execution date is in future: %s", dag_run.execution_date)
             return callback
 
-        self._verify_integrity_if_dag_changed(dag_run=dag_run, session=session)
+        if not self._verify_integrity_if_dag_changed(dag_run=dag_run, session=session):
+            self.log.warning("The DAG disappeared before verifying integrity: %s. Skipping.", dag_run.dag_id)
+            return None
```

Comment:
```suggestion
            return callback
```
I think that's the pattern we use here

Reply:
Right. Thanks :)